filter_plugins/collection.py

```diff
@@ -19,19 +20,20 @@ def is_member_of(value, membership):
     else:
         raise errors.AnsibleFilterError("Provided membership list is not a list!")
 
+
 def dot1x_interfaces(configuration, type='dot1x'):
     """
     Return list of dot1x enabled interfaces.
 
     Default will return all dot1x interfaces, use type to get a subset.
     """
-    ## Redo this!
+    # Redo this!
     dot1x_filter = set(['dot1x', 'dot1x_mab', 'dot1x_parking'])
     if type not in dot1x_filter:
         raise errors.AnsibleFilterError("Invalid type provided. Valid types: dot1x, dot1x_mab, dot1x_parking")
 
     interface_list = []
-    for iface, iface_config in configuration['interfaces'].iteritems():
+    for iface, iface_config in iteritems(configuration['interfaces']):
```

Comment:
You can avoid the compatibilty wrapper here by just using `dict.items()`
Sure it is less efficient in Python 2, but we really don't have _that_ many interfaces to list.